data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -184,20 +182,28 @@ private void incrementErrorCounters(final Exception e) {
     }
 
     public void execute(final AccumulatingBulkRequest bulkRequest) throws InterruptedException {
-        // Exponential backoff run forever
-        // TODO: replace with custom backoff policy setting including maximum interval between retries
-        final BackOffUtils backOffUtils = new BackOffUtils(
-                BackoffPolicy.exponentialBackoff(TimeValue.timeValueMillis(50), Integer.MAX_VALUE).iterator());
+        final Backoff backoff = Backoff.exponential(INITIAL_DELAY_MS, MAXIMUM_DELAY_MS).withMaxAttempts(maxRetries);
```

Comment:
The original behavior of the sink was to have unlimited retries. Should this have some sort of configuration such as the following:

```
Backoff backoff = Backoff.exponential(INITIAL_DELAY_MS, MAXIMUM_DELAY_MS)
if(maxRetries != 0) {
  backoff = backoff.withMaxAttempts(maxRetries);
}
```

Reply:
It appears that the current configuration uses `Integer.MAX_VALUE` to indicate infinite retries.

https://github.com/opensearch-project/data-prepper/blob/738293eb4f182764d2f4e7e3ed2e686ca973cab7/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/RetryConfiguration.java#L43

I think we can hook off of this for this PR. But, we should probably consider a better approach in the future. As it is, you cannot explicitly set infinite retries in the YAML without knowing the maximum value in Java (which is rather opaque for a user in my opinion).

Reply:
max retries is always set to default value of `Integer.MAX_VALUE`

Reply:
The `max_retries` can be set by the user.

What I'm suggesting for this PR is that we don't create a very large limit, but actually make it unlimited. By not calling `withMaxAttempts(int)` there will be no maximum at all.

Backoff backoff = Backoff.exponential(INITIAL_DELAY_MS, MAXIMUM_DELAY_MS)
if(maxRetries != Integer.MAX_VALUE) {
  backoff = backoff.withMaxAttempts(maxRetries);
}